File: Mod06/Democode/CS/EncapsulationDemo_CS/EncapsulationDemo_CS/Person.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EncapsulationDemo_CS
{
    class Person
    {

        private float _weight;
        private float _height;
        private string _ethnicity;
        private string _gender;

        #region Properties
        public float Weight
        {
            get { return _weight; }
            set
            {
                if (value > 0)
                {
                    this._weight = value;
                }
            }
        }

        public float Height
        {
            get { return _height; }
            set
            {
                if (value > 0)
                {
                   this._height = value;
                }
            }
        }

        public string Ethnicity
        {
            get { return _ethnicity; }
            set { this._ethnicity = value; }
        }

        public string Gender
        {
            get { return _gender; }
            set
            {
                if (value == "male" || value == "female")
                {
                    this._gender = value;
                }
            }
        }
        #endregion

        public Person(float weight, float height, string ethnicity, string gender)
        {
            this._weight = weight;
            this._height = height;
            this._ethnicity = ethnicity;
            this._gender = gender;
        }

        
    }
}

File: Mod06/Democode/VB/EncapsulationDemo_VB/EncapsulationDemo_VB/Person.vb
Public Class Person

    Private _weight As Single
    Private _height As Single
    Private _ethnicity As String
    Private _gender As String

    Public Property Weight As Single

        Get
            Return Me._weight
        End Get

        Set(value As Single)
            If value > 0 Then
                Me._weight = value
            End If
        End Set

    End Property

    Public Property Height As Single
        Get
            Return Me._height
        End Get
        Set(value As Single)
            If value > 0 Then
                Me._height = value
            End If
        End Set

    End Property

    Public Property Ethnicity As String
        Get
            Return Me._ethnicity
        End Get
        Set(value As String)
            Me._ethnicity = value
        End Set

    End Property

    Public Property Gender As String
        Get
            Return Me._gender
        End Get
        Set(value As String)
            If value = "male" Or value = "female" Then
                Me._gender = value
            End If
        End Set

    End Property

    Public Sub New(ByVal weight As Single, ByVal height As Single, ByVal race As String, ByVal gender As String)

        Me._weight = weight
        Me._height = height
        Me._ethnicity = race
        Me._gender = gender

    End Sub

End Class
